flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/ProjectionCodeGenerator.scala

```diff
@@ -124,6 +129,93 @@ object ProjectionCodeGenerator {
     new GeneratedProjection(className, code, ctx.references.toArray, ctx.tableConfig)
   }
 
+  /**
+   * If adaptive hash agg takes effect, and the sample result was to suppress local hash agg. In
+   * order to ensure that the data format transmitted downstream with doing local hash agg is
+   * consistent with the data format transmitted downstream without doing local hash agg, we need to
+   * do projection for grouping function value.
+   *
+   * <p> For example, for sql statement "select a, avg(b), count(c) from T group by a", if local
+   * hash agg suppressed and a row (1, 5, "a") comes to local hash agg, we will pass (1, 5, 1, 1) to
+   * downstream.
+   */
+  def generatedAdaptiveHashAggValueProjectionCode(
+      ctx: CodeGeneratorContext,
+      inputType: RowType,
+      outClass: Class[_ <: RowData] = classOf[BinaryRowData],
+      inputTerm: String = DEFAULT_INPUT1_TERM,
+      aggInfos: Array[AggregateInfo],
+      outRecordTerm: String = DEFAULT_OUT_RECORD_TERM,
+      outRecordWriterTerm: String = DEFAULT_OUT_RECORD_WRITER_TERM): String = {
+    val fieldExprs: ArrayBuffer[GeneratedExpression] = ArrayBuffer()
+    aggInfos.map {
+      aggInfo =>
+        aggInfo.function match {
+          case _: SumAggFunction =>
+            fieldExprs += GenerateUtils.generateFieldAccess(
```

Comment:
Assuming the sum field type is bigint, if some element of the field value is null, we emit the result to downstream global agg is default value -1L, whether the final is corrected?